### DISCUSSION TITLE
Dom change Issues after wire:click

### DISCUSSION THREAD

**Discussion by alexsmaia** [livewire/livewire#4092] [1. Help] [answered]
I tried everything in the documentation about the Dom Troubleshooting.

When wire:click is pressed the active classes added to dom from the if statements are removed.

My Code:
```
<li class="nav-group">
    <div wire:key="{{ $title }}" class="nav-group__label @if ($this->checkRoute($route)) active @endif" wire:click="dropToggle">
        <div class="icon ani03s"><span class="flex_center xyz xyz-{{ $icon }}"></span></div>
        <div class="title ani03s">{{ __( $title ) }}</div>
        <span class="xyz xyz-angle-down ani03s">
    </div>
    <ul class="nav-group__sub @if( $dropShow ) show @endif">
        @foreach ($links as $link)
            <li wire:key="{{ $loop->index }}">
                <a class="ani03s @if($this->checkRoute($route)) active @endif" href="{{ route($link['route']) }}">{{ __( $link['title'] ) }}</a>
            </li>
        @endforeach
    </ul>
</li>
```

I don't know what to do more, can anyone help me?

Tks.

**Comment by joshhanley** [comment, marked as answer]
@alexsmaia the issue will be that on the initial request the route will be your actual route, but then when Livewire makes a request, it makes a request to it's own endpoint, so the current route will then be "/livewire/message", hence why the active route stops working.

What you can do though, is in the mount method get the current route, which will be your actual route, and save it to a property, then when you check your route, you can compare it with the saved route property.

Hope this helps!

**Reply by alexsmaia** [reply]
Tks @joshhanley, that resolved the issue :)

**Reply by joshhanley** [reply]
No worries 🙂
